src/Npgsql/Replication/PgOutput/PgOutputAsyncEnumerable.cs

```diff
@@ -59,12 +63,38 @@ sealed class PgOutputAsyncEnumerable : IAsyncEnumerable<PgOutputReplicationMessa
         CancellationToken cancellationToken,
         NpgsqlLogSequenceNumber? walLocation = null)
     {
+        _protocolVersion = options.ProtocolVersion;
         _connection = connection;
         _slot = slot;
         _options = options;
         _baseCancellationToken = cancellationToken;
         _walLocation = walLocation;
 
+
+        if (_protocolVersion >= PgOutputProtocolVersion.V2)
```

Comment:
Yes, that was actually my intent.
Whenever I change/update something replication protocol related I find myself figuring out which version introduced what and am happy about every bit of information I find in the code (PostgreSQL's documentation regarding replication is not perfect and I repetedly have to dig through their code to make sure I get it right).
That's actally also the reason for the enum, because that way I can add helpful documentation comments to each version.